SwissDataScienceCenter/renku-ui
# Synchronize UI with project authorization implementation

# Changes

## Show number of members in the info bar

<img width="184" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/bdbc5684-a3ae-4352-a5ba-b738d54f1863">

## Add members section to settings

<img width="1146" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/3c726671-10bb-4857-9688-dc6bbcf83c49">

## Adapt UI to privileges available to role

### Editor and Viewer can only see members

<img width="187" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/1447ae38-972f-4d6a-9f6c-8f0731ea5a5c">

<img width="1170" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/47073e07-0ddc-464e-9f73-59f652af9f46">

### Editor and Viewer cannot delete projects

<img width="137" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/0e450403-9c7f-4569-916a-7cf92a480dbe">

### Editor can change certain metadata

<img width="1170" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/1ca34c7d-4658-4c85-8e50-fbcbf2bf5f71">


### Viewer cannot add sessions, data sources, or repos

<img width="1342" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/4565ba8c-18b9-4c15-8571-095b0924cec5">

### Viewer cannot edit metadata

<img width="1196" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/df1c88b6-8ad8-4c42-a719-c9a6dbea94e9">


## Testing

- Create two users
- Create a project 1 as user 1, and give user 2 viewer access to the project
- Log in as user 2 and check the user can only view the project
- Create a project 2 as user 1, and give user 2 editor access to the project
- Log in as user 2 and check the user can edit the description and name, but not other metadata
- Check that user 2 can add sessions, data sources, and repositories

/deploy renku=release-0.52.x renku-data-services=build-authorization-pt2
